src/nile/core/declare.py

```diff
@@ -1,24 +1,58 @@
 """Command to declare StarkNet smart contracts."""
 import logging
 
+from starkware.starknet.definitions import constants
+from starkware.starknet.services.api.contract_class import ContractClass
+from starkware.starknet.services.api.gateway.transaction import (
+    DECLARE_SENDER_ADDRESS,
+    Declare,
+)
+
 from nile import deployments
-from nile.common import DECLARATIONS_FILENAME, parse_information, run_command
+from nile.common import (
+    ABIS_DIRECTORY,
+    BUILD_DIRECTORY,
+    DECLARATIONS_FILENAME,
+    get_gateway_response,
+)
 
 
-def declare(contract_name, network, alias=None, overriding_path=None):
+async def declare(
+    contract_name, network, alias=None, signature=None, overriding_path=None, token=None
+):
     """Declare StarkNet smart contracts."""
     logging.info(f"🚀 Declaring {contract_name}")
 
     if alias_exists(alias, network):
         file = f"{network}.{DECLARATIONS_FILENAME}"
         raise Exception(f"Alias {alias} already exists in {file}")
 
-    output = run_command(contract_name, network, overriding_path, operation="declare")
-    class_hash, tx_hash = parse_information(output)
+    base_path = (
+        overriding_path if overriding_path else (BUILD_DIRECTORY, ABIS_DIRECTORY)
+    )
+    artifact = f"{base_path[0]}/{contract_name}.json"
+    open_artifact = open(artifact, "r")
+    contract_class = ContractClass.loads(data=open_artifact.read())
+
+    if signature is None:
+        signature = []
+
+    tx = Declare(
```

Comment:
Why are we sending the transaction at this low level instead of using the [declare](https://github.com/starkware-libs/cairo-lang/blob/0ba3ff59c1c86f2a30adc8fd144eaacb22c48ce9/src/starkware/starknet/cli/starknet_cli.py#L710) function directly?

I think we are trying to address two problems at the same time, the first one being **replacing subprocess calls with direct python calls** (as the name of the PR states), and the second one is **obtaining output in a better format than string** (from print statements).

To solve the first problem, we can import and use the commands from the `starknet.cli` module directly, (like [declare](https://github.com/starkware-libs/cairo-lang/blob/0ba3ff59c1c86f2a30adc8fd144eaacb22c48ce9/src/starkware/starknet/cli/starknet_cli.py#L710)), we don't need to simulate the implementation of that command (as I think we are trying to do with the current code). The current code solves the second problem though, because sadly starknet doesn't provide a Runtime Environment, so to get the output in a better format, we need to "hack" the cli.

Summarizing I think we could replace subprocess more easily by using the CLI commands where we currently call subprocess (I may be missing the full picture). But I think the second problem of course is still super important (could be separated into a different issue maybe?).

Reply:
@ericnordelo I totally agree with the above points^ haha that's why in the new PR (https://github.com/OpenZeppelin/nile/pull/236), we are using the `starknet_cli` directly. I'll close this PR to avoid further confusion. Sorry about that!